### PR TITLE
feat: commit using GitHub API for verified commits

### PR DESCRIPTION
**Pull request by joshjohanning**
Key changes:

**New Features and Configuration:**

* Uses GitHub API to create the commit so that it is verified (but not if committing node modules because that's too many with API rate limits)
* Update the default commit message to `chore: prepare ${version} release` (from `chore: prod dependencies`)
* Added support for a `github_api_url` input to allow targeting GitHub Enterprise Server or custom API endpoints, with appropriate default and documentation updates in `action.yml` and `README.md`. [[1]](diffhunk://#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R11-R14) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R13-R36)
* Updated the `README.md` to document new features, inputs, and commit/tagging behavior, including verified commits, annotated tags, and GHES support.

**Testing Enhancements:**

* Expanded the test suite in `__tests__/index.test.js` to cover:
  - Both API and Git CLI commit/tagging paths, including error handling and edge cases.
  - Custom API URL usage for GHES.
  - Minor/major version tags and release branch logic.
  - Mocking of additional Octokit git methods and file system operations. [[1]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L12-R13) [[2]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R36-R55) [[3]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L56-R71) [[4]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R92) [[5]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R110-R113) [[6]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R126-R135) [[7]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R180-R190) [[8]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R200-R276) [[9]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L284-R597)

**Versioning:**

* Bumped the package version to `2.1.0` in `package.json` to reflect the new features and improvements.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -8,6 +8,10 @@ inputs:
   github_token:
     description: 'Token for the github API'
     required: true
+  github_api_url:
+    description: 'GitHub API URL (e.g., https://api.github.com for GitHub.com or https://ghes.domain.com/api/v3 for GHES)'
+    required: false
+    default: '${{ github.api_url }}'
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply action.yml:14]: no

File: README.md [hunk 2]
```diff
@@ -10,6 +10,30 @@ This action creates a release branch for your GitHub Actions which will be autom
 
 Based on the [tgymnich/publish-github-action](https://github.com/tgymnich/publish-github-action) action, but I wanted further customization and control over the release process (i.e.: adding `ncc` output and not committing `node_modules` directory).
 
+## Features
+
+- 🔐 **Verified Commits** - Uses GitHub API to create verified commits (when `commit_node_modules` is `false`)
+- 🏷️ **Annotated Tags** - Creates proper annotated tags via GitHub API
+- 🌐 **GHES Support** - Works with GitHub Enterprise Server
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply README.md]: good catch, correcting

File: README.md [hunk 3]
```diff
@@ -10,6 +10,30 @@ This action creates a release branch for your GitHub Actions which will be autom
...
+| `npm_package_command`    | Command to build the action                                                                                                                        | No       | `npm run package`       |
+| `commit_node_modules`    | Whether to commit `node_modules` folder. **Note:** When set to `true`, commits will NOT be verified due to API limitations with large file counts. | No       | `false`                 |
+| `commit_dist_folder`     | Whether to commit `dist` folder                                                                                                                    | No       | `true`                  |
+| `publish_minor_version`  | Whether to publish minor version tag (e.g., `v1.2`)                                                                                                | No       | `false`                 |
+| `publish_release_branch` | Whether to publish release branch (e.g., `releases/v1.2.3`)                                                                                        | No       | `false`                 |
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply README.md]: no issue

File: src/index.js [hunk 4]
```diff
@@ -9,20 +9,136 @@ import * as github from '@actions/github';
...
+    // Remove .github folder by not including it in the tree
+    core.info('Excluding .github folder from tree');
+
+    // 4. Create new tree
+    const { data: newTree } = await octokit.rest.git.createTree({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      base_tree: commit.tree.sha,
+      tree: treeItems
+    });
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: good catch.

File: __tests__/index.test.js [hunk 5]
```diff
@@ -147,6 +177,17 @@ describe('Publish GitHub Action', () => {
...
+      // Mock git rev-parse output for getting commit SHA
+      const execImplementation = (cmd, args, options) => {
+        if (cmd === 'git' && args[0] === 'rev-parse' && args[1] === 'HEAD') {
+          if (options && options.listeners && options.listeners.stdout) {
+            options.listeners.stdout(Buffer.from('abc123\n'));
+          }
+        }
+        return Promise.resolve(0);
+      };
+      mockExec.exec.mockImplementation(execImplementation);
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply __tests__/index.test.js:191]: good catch, corrected

File: src/index.js [hunk 6]
```diff
@@ -9,20 +9,133 @@ import * as github from '@actions/github';
...
+      core.info(`Added ${distFiles.length} files from dist folder`);
+    }
+
+    // 4. Create new tree
+    const { data: newTree } = await octokit.rest.git.createTree({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      base_tree: commit.tree.sha,
+      tree: treeItems
+    });
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: good catch...let's try to add this

File: src/index.js [hunk 7]
```diff
@@ -64,24 +227,82 @@ export async function run() {
       await exec.exec('git add -f dist');
     }
 
+    // Remove .github folder from local git (for Git CLI path)
+    // API path also removes .github by marking files for deletion in the tree
     await exec.exec('git rm -r .github');
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: implementing

File: src/index.js [hunk 8]
```diff
@@ -52,6 +209,12 @@ export async function run() {
     ]);
 
     if (npmPackageCommand) {
+      // Clean dist folder before building to remove stale files
+      if (commitDistFolder === 'true') {
+        await exec.exec('rm', ['-rf', 'dist']);
+        core.info('Cleaned dist folder before build');
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js:214]: done

File: README.md [hunk 9]
```diff
@@ -10,6 +10,38 @@ This action creates a release branch for your GitHub Actions which will be autom
...
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply README.md]: i like the pipe but will make it easier to read

File: src/index.js [hunk 10]
```diff
@@ -9,20 +9,177 @@ import * as github from '@actions/github';
...
+
+/**
+ * Recursively get all files in a directory
+ * @param {string} dir - Directory to scan
+ * @returns {Array} Array of file objects with path
+ */
+async function getFilesRecursively(dir) {
+  const { readdirSync, statSync } = await import('fs');
+  const { join } = await import('path');
+
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: good catch

File: src/index.js [hunk 11]
```diff
@@ -6,23 +6,178 @@
...
+    const githubApiUrl = core.getInput('github_api_url', { required: false });
     const npmPackageCommand = core.getInput('npm_package_command', { required: false });
     const commitNodeModules = core.getInput('commit_node_modules', { required: false });
     const commitDistFolder = core.getInput('commit_dist_folder', { required: false });
     const publishMinorVersion = core.getInput('publish_minor_version', { required: false });
     const publishReleaseVersion = core.getInput('publish_release_branch', { required: false });
 
     const context = github.context;
-    const octokit = github.getOctokit(githubToken);
+    const octokit = github.getOctokit(githubToken, { baseUrl: githubApiUrl });
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: overly defensive, but putting some check in there.

File: src/index.js [hunk 12]
```diff
@@ -6,23 +6,178 @@
...
+    const items = readdirSync(currentDir);
+
+    for (const item of items) {
+      const fullPath = join(currentDir, item);
+      const stat = statSync(fullPath);
+
+      if (stat.isDirectory()) {
+        scan(fullPath);
+      } else {
+        files.push({ path: fullPath });
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js:157]: fixing

File: src/index.js [hunk 13]
```diff
@@ -6,23 +6,178 @@
...
+ * @returns {Array} Array of file objects with path
+ */
+function getFilesRecursively(dir) {
+  const files = [];
+
+  function scan(currentDir) {
+    const items = readdirSync(currentDir);
+
+    for (const item of items) {
+      const fullPath = join(currentDir, item);
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: fixing
